Mathlib/Algebra/Module/Submodule/Pointwise.lean

```diff
@@ -161,6 +161,7 @@ instance pointwiseAddCommMonoid : AddCommMonoid (Submodule R M)
   zero_add _ := bot_sup_eq
   add_zero _ := sup_bot_eq
   add_comm _ _ := sup_comm
+  nsmul := letI : Zero (Submodule R M) := ⟨⊥⟩; letI : Add (Submodule R M) := ⟨Sup.sup⟩; nsmulRec
```

Comment:
Doesn't this indicate that it would be better to declare the Zero and Add instances separately?

Reply:
Does it matter if we're using `letI`, because this just inlines everything.

Reply:
I think this behaves exactly the same as what we currently do, though I haven't checked.

Reply:
I've pulled this out as requested.